xdsl/riscv_asm_writer.py

```diff
@@ -0,0 +1,56 @@
+from typing import IO
+from xdsl.ir import Operation
+from xdsl.dialects.riscv import RISCVOp, RegisterType, AnyIntegerAttr
+from xdsl.dialects.builtin import StringAttr, IntegerAttr, ModuleOp
+from xdsl.utils.hints import isa
+
+
+def print_riscv_module(module: ModuleOp, output: IO[str]):
+    for op in module.ops:
+        print_assembly_instruction(op, output)
+
+
+def print_assembly_instruction(op: Operation, output: IO[str]) -> None:
+    # default assembly code generator
+    assert isinstance(op, RISCVOp)
+    assert op.name.startswith("riscv.")
+    instruction_name = op.name[6:]
+    components: list[str] = []
+
+    for result in op.results:
+        assert isinstance(result.typ, RegisterType)
+        assert isinstance(result.typ.data.name, str)
+        components.append(result.typ.data.name)
+
+    for operand in op.operands:
+        assert isinstance(operand.typ, RegisterType)
+        assert isinstance(operand.typ.data.name, str)
+        components.append(operand.typ.data.name)
+
+    if "offset" in op.attributes:
+        label_attr = getattr(op, "offset")
```

Comment:
This was something I was hoping to rewrite, I think it would be better to explicitly switch on the operation types and never `getattr`

Reply:
I'd recommend copying over the RISCVOp implementation from `workshop-all`

Reply:
And putting the code in this function back in there

Reply:
Modulo `getattr`